src/AggregateFunctions/AggregateFunctionHistogram.cpp

```diff
@@ -130,10 +130,9 @@ class AggregateFunctionHistogramData
 
         // Maintain doubly-linked list of "active" points
         // and store neighbour pairs in priority queue by distance
-        UInt32 previous[size + 1];
-        UInt32 next[size + 1];
-        bool active[size + 1];
-        std::fill(active, active + size, true);
+        std::vector<UInt32> previous(size + 1);
+        std::vector<UInt32> next(size + 1);
+        std::vector<bool> active(size + 1, true);
```

Comment:
Is it possible to use std::arrays instead of vectors here not to introduce allocations?

Reply:
i don't think so, since it'd also require size to be known at compile time

Reply:
Of cause, you right